lib/lsp-server/src/lib.rs

```diff
@@ -219,13 +219,22 @@ impl Connection {
 #[cfg(test)]
 mod tests {
     use crossbeam_channel::unbounded;
-    use lsp_types::notification::{Exit, Initialized, Notification};
-    use lsp_types::request::{Initialize, Request};
+    use lsp_types::notification::{Initialized, Notification};
     use lsp_types::{InitializeParams, InitializedParams};
     use serde_json::to_value;
 
     use crate::{Connection, Message, ProtocolError, RequestId};
 
+    #[cfg(not(feature = "bsp"))]
+    static INITIALIZE: &str = "initialize";
+    #[cfg(not(feature = "bsp"))]
+    static EXIT: &str = "exit";
+
+    #[cfg(feature = "bsp")]
+    static INITIALIZE: &str = "build/initialize";
+    #[cfg(feature = "bsp")]
+    static EXIT: &str = "build/exit";
```

Comment:
This is exactly what I don't think we want here. Assume (the unlikely case) that someone has a workspace that both defines and lsp server and a bsp server, with this, they can't use the library for both since enabling bsp will break the lsp version. So it would probably be better to duplicate the relevant entry point parts, for bsp. Though I am unsure how deeply those are rooted into the library

Reply:
You're right. I tried to duplicate some of the functions as you said, in such a way as to have as little copy-paste as possible. Please let me know if this is what you meant, and what you think about this solution.